packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -164,6 +164,18 @@ abstract class StateMachineBase extends Resource implements IStateMachine {
     });
   }
 
+  /**
+   * Grant the given identity permissions to start a synchronous execution of
+   * this state machine.
+   */
+  public grantStartSyncExecution(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: ['states:StartSyncExecution'],
+      resourceArns: [this.stateMachineArn],
+    });
+  }
```

Comment:
Why not just add 'states:StartSyncExecution' to `grantStartExecution()`? Any risk in granting both actions to a state machine?

Even better would be to check the state machine type and grant the correct action.

Users who are very specific that this should be specific can use the generic `grant()` API (from below).

Reply:
> Even better would be to check the state machine type and grant the correct action.

This can potentially break as users won't be able to start express workflows asynchronously.